src/react/utils.js

```diff
@@ -389,6 +392,25 @@ export function flattenChildren(realm: Realm, array: ArrayValue): ArrayValue {
   return flattenedChildren;
 }
 
+export function evaluateComponentTreeBranch(realm: Realm, effects: Effects, nested: boolean, f: Function) {
+  let [
+    value,
+    generator,
+    modifiedBindings,
+    modifiedProperties: Map<PropertyBinding, void | Descriptor>,
+    createdObjects,
+  ] = effects;
+  if (nested) {
+    realm.applyEffects([value, new Generator(realm), modifiedBindings, modifiedProperties, createdObjects]);
```

Comment:
Try `realm.withEffectsApplied` (you may have to rebase)

Reply:
I couldn't see that method on realm?